### PR TITLE
[Bifrost] Init single-node loglets with random loglet ids

### PR DESCRIPTION
**Pull request by AhmedSoliman**
[Bifrost] Init single-node loglets with random loglet ids

In preparation to allow multi-segment chains, local loglets in single-node bootstrap now get unique random ids.

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/restatedev/restate/pull/1720).
* #1726
* #1725
* #1722
* __->__ #1720
* #1718
* #1717
* #1705
* #1704

### CODE REVIEW

File: crates/types/src/logs/metadata.rs [hunk 1]
```diff
@@ -235,18 +248,43 @@ impl Chain {
     }
 }
 
+/// Creates appropriate [[LogletParams]] value that can be used to start a fresh
```
Comment (tillrohrmann) [inline crates/types/src/logs/metadata.rs:251]: Is there a difference between `[[LogletParams]]` and ``[`LogletParams`]``? So far we mostly used the latter in Rust docs, I believe.

File: crates/types/src/logs/metadata.rs [hunk 2]
```diff
@@ -235,18 +248,43 @@ impl Chain {
...
     // Get metadata from somewhere
     let mut builder = LogsBuilder::default();
-
+    #[allow(clippy::mutable_key_type)]
+    let mut generated_params: HashSet<_> = HashSet::new();
     // pre-fill with all possible logs up to `num_partitions`
     (0..num_partitions).for_each(|i| {
-        // fixed config that uses the log-id as loglet identifier/config
-        let config = LogletParams::from(i.to_string());
+        // a little paranoid about collisions
```
Comment (tillrohrmann) [inline crates/types/src/logs/metadata.rs:278]: What are the odds ;-) I guess it's the birthday problem.

**Review by tillrohrmann — APPROVED** [review]
Nice work @AhmedSoliman. LGTM. +1 for merging.

File: crates/types/src/logs/metadata.rs (hunk 1 above)
Comment (AhmedSoliman) [reply crates/types/src/logs/metadata.rs:251]: No difference, will fix to make it consistent.

**Pull request merged**